charts/shoot-core/components/charts/kube-proxy/templates/kube-proxy-daemonset.yaml

```
@@ -45,6 +46,26 @@ spec:
           /sbin/sysctl -w net.ipv6.conf.all.disable_ipv6=1 || echo "IPv6 is already disabled. Doing nothing..."
         securityContext:
           privileged: true
+{{- end }}
+{{- if (semverCompare ">= 1.16" .Values.kubernetesVersion) }}
+      initContainers:
+      - name: kube-proxy-cleanup
+        image: {{ index .Values.images "kube-proxy" }}
+        env:
+        - name: KUBE_PROXY_MODE
+          value: {{ include "kube-proxy.mode" . }}
+        imagePullPolicy: IfNotPresent
+        command: ['sh', '-c', "/script/init.sh"]
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: kube-proxy-init-script
+          mountPath: /script/init.sh
+          subPath: init.sh
```

```suggestion
          mountPath: /script
```

It would be even better if you mount it under `/var/lib/kube-proxy/script`